sympy/tensor/array/expressions/array_expressions.py

```diff
@@ -1458,6 +1459,35 @@ def as_explicit(self):
         return tensorcontraction(self.expr.as_explicit(), *self.contraction_indices)
 
 
+class Reshape(_CodegenArrayAbstract):
+
+    def __new__(cls, expr, shape):
+        expr = _sympify(expr)
+        if not isinstance(shape, Tuple):
+            shape = Tuple(*shape)
+        obj = Expr.__new__(cls, expr, shape)
+        obj._shape = tuple(shape)
+        obj._expr = expr
+        return obj
+
+    @property
+    def shape(self):
+        return self._shape
+
+    @property
+    def expr(self):
+        return self._expr
+
+    def as_explicit(self):
+        ee = self.expr.as_explicit()
+        if isinstance(ee, MatrixCommon):
+            from sympy import Array
+            ee = Array(ee)
+        elif isinstance(ee, MatrixExpr):
+            return self
```

Comment:
Why would `ee` still be `MatrixExpr` here (after the `as_explicit` call above)?

Reply:
The shape of the matrix can be symbolic. In such case, `.as_explicit()` will not evaluate.